### PR TITLE
LRDOCS-4930, Advanced Form Features

### PR DESCRIPTION
**Pull request by rbohl**
https://issues.liferay.com/browse/LRDOCS-4930. This is going to include a lot of small articles. I'm sending it now, with the first three drafts completed, for autocomplete, translation, and autosave. I will continue adding commits with new articles, and at the end I will go back to the beginning to make edits from your review. Does that make sense?

### CODE REVIEW

**Comment by natocesarrego** [comment]
No problem for me, @rbohl. However, do you mind to warn me when you do the last commit? Then, I'll know that this article can be reviewed. 🙂

**Comment by rbohl** [comment]
@natocesarrego Ready for SME Review!

**Comment by natocesarrego** [comment]
Thanks for the update, @rbohl!

**Comment by natocesarrego** [comment]
Starting review!

**Comment by rbohl** [comment]
Thanks! Feel free to review this PR or to just look at what's in master since it went through the KM review process already

**Comment by natocesarrego** [comment]
Thanks for the warning, Russell!

I'm reviewing using master to avoid review outdated information. 🙂 

Best.

File: discover/portal/articles/130-forms/10-success-pages.markdown [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+1.  Add a form in Site Administration &rarr; Content &rarr; Forms. 
+
+2.  Click on the form page's Actions
+    (![Actions](../../images-dxp/icon-actions.png)) button and choose *Add
+    Success Page*.
+
+![Figure x: Add a success page using the edit menu for the form page.](../../images/forms-success-page-add.png)
+
+Once the success page is added to your form, fill in the Title and Description
```
Comment (natocesarrego) [inline discover/portal/articles/130-forms/10-success-pages.markdown:37]: Actually, the fields used to configure a _Success page_ are named _Title_ and _Content_. 😉

File: discover/portal/articles/130-forms/14-validation.markdown [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+conditions are for comparing the value of the number entered to some other
+value.
+
+Available conditions to check include:
+
+- Is greater than or equal to
+- Is greater than
+- Is not equal to
+- Is less than or equal to
+- Is less than or equal to
```
Comment (natocesarrego) [inline discover/portal/articles/130-forms/14-validation.markdown:83]: Duplicated line.

File: discover/portal/articles/130-forms/14-validation.markdown [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+![Figure x: Make sure User-entered numeric data is within reasonable bounds.](../../images/forms-numeric-val1.png)
+
+To make sure Users don't enter a number over 10, make sure the enable validation
+and use the *Is greater than or equal to* condition with a value of *10*. Use
+the message *Please enter 10 or less**.
+
+![Figure x: Numeric conditions constrain User-entered numeric data.](../../images/forms-numeric-val2.png)
+
+Numeric fields are themselves just Text Fields validated to only allow numeric
+data entry. That's why they're in the Customized Elements section of the Form
```
Comment (natocesarrego) [inline discover/portal/articles/130-forms/14-validation.markdown:95]: Nice explanation regarding the _Customized Elements_ and _My numeric type is_! 👏

File: discover/portal/articles/130-forms/16-form-notifications.markdown [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+       (for example, *test@liferay.com*). 
+    -  Subject: An informative subject line tells the recipient what happened:
+       *An application for employment was submitted in The Lunar Resort*, for
+       example.
+
+![Figure x: Configure email notifications each time a form entry is submitted.](../../images/forms-notification-email.png)
+
+If you 
+[enabled workflow for the form](/discover/portal/-/knowledge_base/7-1/workflow-and-forms),
+and there's already a notification going to the people who need to process the
```
Comment (natocesarrego) [inline discover/portal/articles/130-forms/16-form-notifications.markdown:25]: Nice note! 👏

File: discover/portal/articles/130-forms/18-form-permissions.markdown [hunk 5]
```diff
@@ -0,0 +1,37 @@
+# Form Permissions
+
+By default, you can grant the following permissions for a form:
+
+-  Delete: Permission to delete the form.
+-  Permissions: This gives any role with the permission the ability to configure
+   the form's permissions.
+-  Add Record: Permission to submit form entries.
```
Comment (natocesarrego) [inline discover/portal/articles/130-forms/18-form-permissions.markdown:8]: It was renamed on the UI to _Add Form Instance Record_.

File: discover/portal/articles/130-forms/18-form-permissions.markdown [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+-  Update: Permission to update form entries.
+-  View: All the other permissions you can configure for the form depend on this
+   one. Think about it: if you can't see the form, how can you add a record?
+
+![Figure 14: You can configure the permissions for a form.](../../images/forms-form-permissions.png)
+
+By default, unauthenticated visitors to your site (Guests) can view and fill out
+forms, because the *Guest* role has *View* and *Add Record* permissions.
```
Comment (natocesarrego) [inline discover/portal/articles/130-forms/18-form-permissions.markdown:16]: It was renamed on the UI to _Add Form Instance Record_.

File: discover/portal/articles/130-forms/18-form-permissions.markdown [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+Guest role the View and Add permissions, then click *Save*.
+
++$$$
+
+**Note:** By default, all users inherit the Guest role's permissions. The Guest
+role represents unauthenticated visitors of your site, and it makes sense that
+if you want to allow Guest users to submit forms, you're fine with your site
+members and portal users submitting forms, as well. If you want to disable the
+automatic inheritance of the Guest role permissions, there's a
+[property](@platform-ref@/7.1-latest/propertiesdoc/portal.properties.html#Permissions)
```
Comment (natocesarrego) [inline discover/portal/articles/130-forms/18-form-permissions.markdown:30]: On master, this link was changed to _thisproperty_, hence a blank space is missing to separate the words. 😉

File: discover/portal/articles/130-forms/19-styling-forms.markdown [hunk 8]
```diff
@@ -0,0 +1,112 @@
+# Styling Form Pages
+
+Use form rows and columns to save vertical space and make the form
+user-friendly. 
+
+A separate article shows you how to use 
+[form pages](/discover/portal/-/knowledge_base/7-1/form-pages).
+This article shows you how to lay out those pages.
+
+Two features are available for creating form layouts:
```
Comment (natocesarrego) [inline discover/portal/articles/130-forms/19-styling-forms.markdown:10]: On master, _are_ is duplicated.

**Review by natocesarrego — COMMENTED** [review]
Review completed, @rbohl!

**Comment by rbohl** [comment]
Review edits forwarded: https://github.com/sez11a/liferay-docs/pull/3481

**Pull request merged**